### ISSUE TITLE
[bug] False positive for "@eslint-react/dom/no-missing-button-type"

### ISSUE THREAD

**Issue by milang** [Rel1cx/eslint-react#1247]
### Describe the bug

When specifying non-trivial expression as value of `type` attribute in `<button type={...}>`, ESLint reports `@eslint-react/dom/no-missing-button-type` issue:

<img width="625" height="208" alt="Image" src="https://github.com/user-attachments/assets/a113a979-bcff-4f28-8303-51086ea39ad7" />

The `type` attribute is specified, yet the rule claims it is missing.

### Reproduction

Vite template project reproducing the issue: [react-button-issue.zip](https://github.com/user-attachments/files/22607876/react-button-issue.zip)

### Expected behavior

Rule should not be reporting the issue when `type` attribute is specified.

### Platform and versions

```Text
node@22.20.0, yarn@4.10.3, @eslint-react/eslint-plugin@2.0.2.
```
